opentracing/tracecontext.go

```diff
@@ -132,3 +144,14 @@ type TraceContextSource interface {
 	// to NewRootTraceContext().
 	NewRootTraceContext() TraceContext
 }
+
+var kTraceTagRegexp = regexp.MustCompile("^(?i:[a-z0-9][-a-z0-9]*)$")
+
+// Returns the canonicalized version of trace tag key `key`, and true if and
+// only if the key was valid.
+func CanonicalizeTraceTagKey(key string) (string, bool) {
+	if !kTraceTagRegexp.MatchString(key) {
```

Comment:
The regexp starts with `(?i:` which specifies case-insensitive matching...

I hear you on caching, though I don't want to prematurely optimize...